core/src/com/unciv/logic/map/mapgenerator/NaturalWonderGenerator.kt

```diff
@@ -180,59 +177,79 @@ class NaturalWonderGenerator(val ruleset: Ruleset, val randomness: MapGeneration
 
     companion object {
         fun placeNaturalWonder(wonder: Terrain, location: Tile) {
-            clearTile(location)
             location.naturalWonder = wonder.name
-            if (wonder.turnsInto != null)
+            if (wonder.turnsInto != null) {
+                clearTile(location)
                 location.baseTerrain = wonder.turnsInto!!
-
-            var convertNeighborsExcept: String? = null
-            var convertUnique = wonder.getMatchingUniques(UniqueType.NaturalWonderConvertNeighbors).firstOrNull()
-            var convertNeighborsTo = convertUnique?.params?.get(0)
-            if (convertNeighborsTo == null) {
-                convertUnique = wonder.getMatchingUniques(UniqueType.NaturalWonderConvertNeighborsExcept).firstOrNull()
-                convertNeighborsExcept = convertUnique?.params?.get(0)
-                convertNeighborsTo = convertUnique?.params?.get(1)
+            } else {
+                clearTile(location, wonder.occursOn)
             }
 
-            if (convertNeighborsTo != null) {
-                for (tile in location.neighbors) {
-                    if (tile.baseTerrain == convertNeighborsTo) continue
-                    if (tile.baseTerrain == convertNeighborsExcept) continue
-                    if (convertNeighborsTo == Constants.coast) {
-                        for (neighbor in tile.neighbors) {
-                            // This is so we don't have this tile turn into Coast, and then it's touching a Lake tile.
-                            // We just turn the lake tiles into this kind of tile.
-                            if (neighbor.baseTerrain == Constants.lakes) {
-                                neighbor.baseTerrain = tile.baseTerrain
-                                neighbor.setTerrainTransients()
-                            }
-                        }
-                        location.setConnectedByRiver(tile, false)
+            val conversionUniques = wonder.getMatchingUniques(UniqueType.NaturalWonderConvertNeighbors, StateForConditionals.IgnoreConditionals) +
+                wonder.getMatchingUniques(UniqueType.NaturalWonderConvertNeighborsExcept, StateForConditionals.IgnoreConditionals)
+            if (conversionUniques.none()) return
+
+            for (tile in location.neighbors) {
+                val state = StateForConditionals(tile = tile)
+                for (unique in conversionUniques) {
+                    if (!unique.conditionalsApply(state)) continue
+                    val index = if (unique.type == UniqueType.NaturalWonderConvertNeighborsExcept) {
+                        if (tile.matchesWonderFilter(unique.params[0])) continue
+                        1
+                    } else 0
+                    val convertTo = unique.params[index]
```

Comment:
I don't like this at all, put the "unique.params[index]" in the if, it has no meaning outside of it
```suggestion
                    val convertTo = if (unique.type == UniqueType.NaturalWonderConvertNeighborsExcept) {
                        if (tile.matchesWonderFilter(unique.params[0])) continue
                        unique.params[1]
                    } else unique.params[0]
```